### ISSUE TITLE
Database design and initialization

### ISSUE THREAD

**Issue by qy21gafy** [qy21gafy/AMOSSharedDeskFeatureBoard#6]
## User story
1. As a user
2. I want / need my registration data, booked desks and added resources to the system to be stored in a database
3. So that they don't get lost and are visible to other users

## Acceptance criteria
* Database design should consider user data (name, company, permissions, admin status etc.)
* Database design should consider booked desks of each user (desk, time, place, user etc.)
* Database design should consider available resources of each company (name, type, place, capacity etc.)
* Database design should consider different company data (e.g name, country, locations, buildings etc.)
* Database design should not contain a lot of redundant data and should be easily extensible
* After finishing a design, a prototyp with dummy data should be set up to allow easy testing
* SQL Database should be used, industry partner suggested MS-SQL-Server Express or PostgreSQL

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by xilef45** [comment]
![Image](https://user-images.githubusercontent.com/88111260/198528165-5655070e-ff9b-45dd-ab97-7b0b44994546.jpg)

**Comment by Ardianpalloshi** [comment]
Hi, this looks good. Do you think this is something you would like to discuss with our indrustry partner in tuesday in regards to authentication? Maybe he can share some more details.

**Comment by Ardianpalloshi** [comment]
I am not sure if this answers the question, or makes sense in this step. But any person which has a username with the domain of the company should be able to book, restriction would be that if the domain does not match then it is not possible to book. Meaning, that when you add the company in the first step, all users need to use the company email address to register and book tables

**Comment by n3rdc4ptn** [comment]
For me the sketch looks also good.
Do you discussed about creating an ER diagram? Such a diagram could be perfect for our documentation and to look at it when working on our code.
For the authentication we can maybe add user groups inside a company? So a user can be part of a user group and a user group has a list of rooms and buildings which they can access? Or we leave it out for now to have a more minimal prototype in the beginning which has no permission management for the users and every user can book every desk in its company?

**Comment by qy21gafy** [comment]
Maybe we could also add an "isApproved" or "awaitingApproval" bool property to the user for the registration/login function.
